rust/lance/src/index/vector/pq.rs

```diff
@@ -318,6 +391,29 @@ impl VectorIndex for PQIndex {
             metric_type: self.metric_type,
         }))
     }
+
+    fn check_can_remap(&self) -> Result<()> {
+        Ok(())
+    }
+
+    fn remap(&mut self, mapping: &HashMap<u64, Option<u64>>) -> Result<()> {
+        let remapped_ids =
+            UInt64Array::from_iter_values(self.row_ids.as_ref().unwrap().values().iter().map(
+                |old_row_id| {
+                    mapping
+                        .get(old_row_id)
+                        .cloned()
+                        // If the row id is not in the mapping then this row is not remapped and we keep as is
+                        .unwrap_or(Some(*old_row_id))
+                        // If the row is in the mapping, but maps to None, then it is deleted, and we insert
```

Comment:
If the row is deleted , let's just not throw it away?  IIUC, putting a tombstone in index does not help any process later to clean up this row?